starfish/pipeline/features/pixels/pixel_spot_detector.py

```diff
@@ -34,132 +28,31 @@ def __init__(
             spots with intensity less than this value are filtered (default 1)
         area_threshold : int
             spots with total area less than this value are filtered
-        crop_size : int
-            ??? # TODO ambrosejcarr what is this?
+        crop_size : Tuple[int, int, int]
+            number of pixels to clip around the border of the image, default = (0, 0, 0)
 
         """
-        self.codebook = pd.read_csv(codebook, dtype={'barcode': object})
-        self.weighted_codes: pd.DataFrame = self._normalize_barcodes(self.codebook)
+        if isinstance(codebook, str):
+            self.codebook = pd.read_json(codebook)
+        else:
+            self.codebook = codebook
         self.distance_threshold = distance_threshold
         self.magnitude_threshold = magnitude_threshold
         self.area_threshold = area_threshold
         self.crop_size = crop_size
 
-        self.label_image: Optional[np.ndarray] = None
-        self.decoded_image: Optional[np.ndarray] = None
-
-    @staticmethod
-    def encode(stack: Stack) -> pd.DataFrame:
-
-        sq = stack.image.squeeze()
-        tile_metadata = stack.image.tile_metadata
-        num_bits = int(tile_metadata['barcode_index'].max() + 1)
-
-        # linearize the pixels, mat.shape = (n_hybs * n_channels * n_z_slice, x * y)
-        mat = np.reshape(sq.copy(), (sq.shape[0], sq.shape[1] * sq.shape[2]))
-
-        res = pd.DataFrame(mat.T)
-        res['spot_id'] = range(len(res))
-        res = melt(
-            df=res,
-            new_index_name='barcode_index',
-            new_value_name='intensity',
-            melt_columns=range(num_bits)
+    def find(self, stack: ImageStack) \
+            -> Tuple[IntensityTable, List, np.ndarray, np.ndarray, np.ndarray]:
+        pixel_intensities = stack.to_pixel_intensities(crop=self.crop_size)
+        decoded_intensities = self.codebook.decode_euclidean(
+            pixel_intensities,
+            max_distance=self.distance_threshold,
+            min_intensity=self.magnitude_threshold
```

Comment:
i'm not sure this is working as intended -- let's trace through the use of this parameter on a whiteboard to make certain.